### PR TITLE
2022.26.12 updates

### LINKED ISSUE #32: Missing package dependencies in setup.cfg
The SDK has required external dependencies to run within a Python application. However, these dependencies are not referenced in `setup.cfg`. This can result in a client installing the package without installing required package dependencies first.

https://github.com/Lockstep-Network/lockstep-sdk-python/blob/34a2551fdad8b50c0c80de65fe76aaf2ed47893e/setup.cfg#L17-L21

* Add required package dependencies to `setup.cfg` options - [Reference](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#declaring-required-dependency)
  * requests
  * dataclasses;python_version<'3.7'

### PR DESCRIPTION
**Pull request by AlexRussak**
Closes #32

### CODE REVIEW

File: src/lockstep/models/actionresultmodel.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    Represents the result of an action.
+
+    In the Lockstep API, an Action is returned when an API call does not
+    produce any data but does produce messages that can be useful in
+    understanding what work was performed. You may use the messages text
+    to display user visible error messages or the results of various
+    operations.
+    """
+
+    messages: list[str] | None = None
```
Comment (sfwatanabe) [inline src/lockstep/models/actionresultmodel.py:29]: I noticed the our setup.cfg indicating support for >= 3.6 but the use of the union pipe operator just got added in 3.10. I can confirm that trying to run this locally causes problems with 3.6. I will follow up with suggestions on the main generator. In the current state we would need to change the min version requirement to 3.10

**Review by sfwatanabe — CHANGES_REQUESTED** [review]
The `__init__.py` files in both clients and models appear to have gotten removed in this PR. Python needs these to detect available imports during runtime. 

Similar to an `index.ts` you'll see in TypeScript projects although they don't require any code.

**Comment by AlexRussak** [comment]
> The `__init__.py` files in both clients and models appear to have gotten removed in this PR. Python needs these to detect available imports during runtime.
> 
> Similar to an `index.ts` you'll see in TypeScript projects although they don't require any code.

Fixed!

**Comment by sfwatanabe** [comment]
In order to support the type hinting union operator `|`, we need to bump the minimum python version to 3.10 in [setup.cfg](https://github.com/Lockstep-Network/lockstep-sdk-python/blob/main/setup.cfg). 

Attempting to install the package currently would only run for 3.10 or higher. We could update the syntax generation to support down to 3.7 in a future PR.

### General type hint changes if below 3.10:
We would need to import from the typing module for hint support. Lower case type hinting for built-in collections wasn't added until 3.9+

```py
# 3.9+
some_numbers: list[int]

#< 3.9
from typing import List
...
some_numbers: List[int]

```
---

### Nullable Attributes
We could wrap everything in an optional
```py
# Before
number: int | None = None

# After
from typing import Optional
...
number: Optional[int] = None
```

Another future option could be to swap over to using a model wrapping library such as [pydantic](https://github.com/pydantic/pydantic).


Additionally from a convenience standpoint we should declare the required dependencies - [related issue](https://github.com/Lockstep-Network/lockstep-sdk-python/issues/32)

**Review by sfwatanabe — APPROVED** [review]

**Pull request merged**